### PR TITLE
fix sporadic failures to load DLLs on init, via lazy loading

### PR DESCRIPTION
**Pull request by AaronFriel**
I've observed that, albeit with low probability, the `init` function in this module causes a crash on Windows. The panic is always of the form:

```
panic: Failed to load user32.dll: A dynamic link library (DLL) initialization routine failed.
goroutine 1 [running]:
syscall.MustLoadDLL(...)
	/opt/hostedtoolcache/go/1.19.0/x64/src/syscall/dll_windows.go:118
github.com/xanzy/ssh-agent.winAPI({0xf395b9?, 0xc00008c900?}, {0xf398ca, 0xb})
	/home/runner/go/pkg/mod/github.com/xanzy/ssh-agent@v0.3.0/pageant_windows.go:68 +0xb4
github.com/xanzy/ssh-agent.init()
	/home/runner/go/pkg/mod/github.com/xanzy/ssh-agent@v0.3.0/pageant_windows.go:62 +0x1ac
```

This panic appears to occur on start of the process, due to the synchronous initialization of DLLs.  This has affected CI tests on the repo https://github.com/pulumi/pulumi where, due to the large number of CI tests running on Windows, we encountered it fairly consistently. We tested using a fork to use the changes in this PR about a month ago (https://github.com/pulumi/pulumi/pull/10207) we haven't seen the issue since. Lazily loading these DLLs reduces memory usage and appears to prevent this panic from occuring.

We'd love to get this merged upstream so that we can update our go.mod dependencies and pass that benefit on to others. As using a replace directive that only applies to our own repo's CI.

### CODE REVIEW

File: pageant_windows.go [hunk 1]
```diff
@@ -17,6 +17,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
+//go:build windows
```
Comment (svanharmelen) [inline pageant_windows.go:20]: Guess we should also add a Go 1.17 style build tag here to make it complete? https://github.com/xanzy/ssh-agent/blob/master/sshagent.go#L17

**Review by svanharmelen — COMMENTED** [review]
Thanks for this one @AaronFriel! Have only one question, rest looks nice 👍🏻

File: pageant_windows.go (hunk 1 above)
Comment (AaronFriel) [reply pageant_windows.go:20]: Fixed!

**Comment by AaronFriel** [comment]
@svanharmelen thanks! Submitted the build tag change, 🙏 looking forward to this merging and getting tagged.

**Review by svanharmelen — APPROVED** [review]
Thanks @AaronFriel 👍🏻 LGTM!

**Pull request merged**